hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -616,4 +616,129 @@ public static function getCurrentContract($contactID)  {
     }
     return null;
   }
+
+  /**
+   * Permanently delete all contracts for given contact ID.
+   *
+   * @param int $contactId
+   *
+   * @return boolean
+   * @throw Exception
+   */
+  public static function deleteAllContractsPermanently($contactId) {
+
+    if (empty($contactId)) {
+      throw new Exception('Please specify contact ID.');
+    }
+
+    $contract = new self();
+    $contract->contact_id = $contactId;
+    $contract->find();
+    while ($contract->fetch()) {
+      self::deleteContractPermanently($contract->id);
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Permanently delete whole contract with its all revisions and entities.
+   *
+   * @param int $contractId
+   *
+   * @return bool
+   * @throw Exception
+   */
+  public static function deleteContractPermanently($contractId) {
```

Comment:
Do you really need this method to be public? It doesn't seem to be used anywhere else outside this class